cmd/lambda/webhook/handler.go

```diff
@@ -2,43 +2,200 @@ package main
 	"go.uber.org/zap"
 )
 
 func main() {
-	l, err := buildHandler()
-	if err != nil {
-		panic(err)
+
+	if os.Getenv("LOCAL_WEBHOOK") == "true" {
+		s, err := buildLocalHandler()
+		if err != nil {
+			panic(err)
+		}
+		err = s.Run(context.Background())
+		if err != nil {
+			panic(err)
+		}
+
+	} else {
+		l, err := buildHandler()
+		if err != nil {
+			panic(err)
+		}
+
+		lambda.Start(l.Handler)
 	}
 
-	lambda.Start(l.Handler)
 }
 
 func buildHandler() (*Lambda, error) {
-	log, err := logger.Build("info")
+	ctx := context.Background()
+	var cfg Config
+	err := envconfig.Process(ctx, &cfg)
+	if err != nil {
+		return nil, err
+	}
+	log, err := logger.Build(cfg.LogLevel)
 	if err != nil {
 		return nil, err
 	}
 	zap.ReplaceGlobals(log.Desugar())
 
+	s, err := NewServer(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+
 	l := Lambda{
-		Server: Server(),
+		Server: s.Routes(),
 	}
 	return &l, nil
 }
 
-func Server() http.Handler {
+type Config struct {
+	LogLevel    string `env:"LOG_LEVEL,default=info"`
+	DynamoTable string `env:"APPROVALS_TABLE_NAME,required"`
+}
+
+type Server struct {
+	db *ddb.Client
+}
+
+func NewServer(ctx context.Context, cfg Config) (*Server, error) {
+	db, err := ddb.New(ctx, cfg.DynamoTable)
+	if err != nil {
+		return nil, err
+	}
+	s := Server{
+		db: db,
+	}
+	return &s, nil
+}
+
+func ValidateToken(ctx context.Context, token string, tokenReq storage.GetAccessTokenByToken) error {
+	if token == tokenReq.Token {
+		//validate token
+		if tokenReq.Result.Start.Equal(tokenReq.Result.End) {
+			return fmt.Errorf("grant start and end time cannot be equal")
+		}
+		if tokenReq.Result.Start.After(tokenReq.Result.End) {
+			return fmt.Errorf("grant start time must be earlier than end time")
+		}
+
+		now := time.Now()
+		if tokenReq.Result.End.Before(now) {
+			return fmt.Errorf("grant finish time is in the past")
+
+		}
+
+	} else {
+
+		return fmt.Errorf("invalid token provided")
+
+	}
+	return nil
+}
+
+func (s *Server) Routes() http.Handler {
 	r := chi.NewRouter()
 	r.Post("/webhook/v1/slack/interactivity", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
+
+	r.Post("/webhook/v1/access-token", func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		t := r.Header.Get("X-Granted-Request")
+
+		//lookup token in database
+		q := storage.GetAccessTokenByToken{Token: t}
+
+		_, err := s.db.Query(ctx, &q)
+		if err != nil {
+			apio.Error(ctx, w, err)
+			return
+		}
+
+		//validate token
+		err = ValidateToken(ctx, t, q)
+		if err != nil {
+			apio.Error(ctx, w, err)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+
+	})
+
+	r.Post("/webhook/v1/test-setup", func(w http.ResponseWriter, r *http.Request) {
+
+		//successfull connection to webhook url return OK
+		w.WriteHeader(http.StatusOK)
+
+	})
+
+	r.Post("/webhook/v1/events-recorder", func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		token := r.Header.Get("X-Granted-Request")
```

Comment:
let's change this - the original `X-Granted-Request` contained the request ID which we've now shifted away from to Access Tokens.

```suggestion
		token := r.Header.Get("X-CommonFate-Access-Token")
```

Reply:
Note that this will require updating the Python library